src/LLVM_Runtime_Linker.cpp

```diff
@@ -1048,11 +1048,6 @@ std::unique_ptr<llvm::Module> get_initial_module_for_ptx_device(Target target, l
         if (!f.isDeclaration() && !f.hasFnAttribute(llvm::Attribute::NoInline)) {
             f.setLinkage(llvm::GlobalValue::AvailableExternallyLinkage);
         }
-
```

Comment:
I think this was the red flag tickling the back of my mind regarding addresses of barriers. It's not being shared that's the problem (via not getting inlined) - it's being duplicated.